core/src/main/java/org/elasticsearch/index/mapper/fixed/FixedPointFieldMapper.java

```diff
@@ -0,0 +1,399 @@
+                    context.allEntries().addText(fieldType().name(), fieldType().nullValueAsString(), boost);
+                }
+            } else if (parser.currentToken() == XContentParser.Token.START_OBJECT
+                && Version.indexCreated(context.indexSettings()).before(Version.V_3_0_0)) {
+                XContentParser.Token token;
+                String currentFieldName = null;
+                Double objValue = fieldType().nullValue();
+                while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                    if (token == XContentParser.Token.FIELD_NAME) {
+                        currentFieldName = parser.currentName();
+                    } else {
+                        if ("value".equals(currentFieldName) || "_value".equals(currentFieldName)) {
+                            if (parser.currentToken() != XContentParser.Token.VALUE_NULL) {
+                                objValue = parser.doubleValue(coerce.value());
+                            }
+                        } else if ("boost".equals(currentFieldName) || "_boost".equals(currentFieldName)) {
+                            boost = parser.floatValue();
+                        } else {
+                            throw new IllegalArgumentException("unknown property [" + currentFieldName + "]");
+                        }
+                    }
+                }
+                if (objValue == null) {
+                    // no value
+                    return;
+                }
+                value = objValue;
+            } else {
+                value = parser.doubleValue(coerce.value());
+                if (context.includeInAll(includeInAll, this)) {
+                    context.allEntries().addText(fieldType().name(), parser.text(), boost);
+                }
+            }
+        }
+        final long scaledValue = (long)(value * (double)fieldType().decimalFactor);
+        if (fieldType().indexOptions() != IndexOptions.NONE || fieldType().stored()) {
+            CustomFixedPointNumericField field = new CustomFixedPointNumericField(scaledValue, fieldType().decimalFactor, fieldType());
+            field.setBoost(boost);
+            fields.add(field);
+        }
+        if (fieldType().hasDocValues()) {
+            addDocValue(context, fields, scaledValue);
+        }
+    }
+
+    @Override
+    protected String contentType() {
+        return CONTENT_TYPE;
+    }
+
+    @Override
+    protected void doXContentBody(XContentBuilder builder, boolean includeDefaults, Params params) throws IOException {
+        super.doXContentBody(builder, includeDefaults, params);
+
+        if (includeDefaults || fieldType().numericPrecisionStep() != Defaults.PRECISION_STEP_64_BIT) {
+            builder.field("precision_step", fieldType().numericPrecisionStep());
+        }
+        if (includeDefaults || fieldType().nullValue() != null) {
+            builder.field("null_value", fieldType().nullValue());
```

Comment:
If we do use `ParseField` in the parsing logic above, we can use `ParseField.getPreferredName()` here to ensure the output XContent always uses the preferred names for parsing.